dataactvalidator/filestreaming/csvAbstractReader.py

```diff
@@ -67,25 +66,54 @@ def openFile(self,region,bucket,filename,csvSchema,bucketName,errorFilename):
             raise ResponseException("Error in header row: CSV file must use only '|' or ',' as the delimiter", StatusCode.CLIENT_ERROR, ValueError, ValidationError.headerError)
 
         self.delimiter = "|" if line.count("|") != 0 else ","
-        for row in csv.reader([line],dialect='excel', delimiter=self.delimiter):
-            for cell in row :
-                headerValue = FieldCleaner.cleanString(cell)
-                if( not headerValue in possibleFields) :
+
+        # check to see if header contains long or short column names
+        # TODO: is there a better way to do this?
+        validation_db = ValidatorValidationInterface()
+        longNameDict = validation_db.getLongToShortColname()
+        headerNames = line.split(self.delimiter)
+        colMatches = 0
+        for i, value in enumerate(headerNames):
+            if value in longNameDict:
+                colMatches += 1
+        if colMatches > 15:
+            longHeaders = True
+        else:
+            longHeaders = False
+
+        # Set the list of possibleFields, converting long col names to their
+        # shorter, machine-readable counterparts that are used in the
+        # staging tables
+        possibleFields = {}
+        for schema in csvSchema:
+            possibleFields[FieldCleaner.cleanString(schema.name_short)] = 0
+
+        for row in csv.reader([line], dialect='excel', delimiter=self.delimiter):
+            for cell in row:
+                submittedHeaderValue = FieldCleaner.cleanString(cell)
+                if longHeaders and submittedHeaderValue in longNameDict:
+                    headerValue = FieldCleaner.cleanString(longNameDict[submittedHeaderValue])
+                elif longHeaders:
+                    headerValue = None
+                else:
+                    headerValue = submittedHeaderValue
+                if not headerValue in possibleFields:
                     # Allow unexpected headers, just mark the header as None so we skip it when reading
                     self.headerDictionary[(current)] = None
```

Comment:
So this is probably a Python thing I'm not getting, and I realize you didn't write this line, but is this making the index actually a single-element tuple?  Is there a reason we're doing this instead of just [current]?